lib/ide-haskell-repl-view.coffee

```diff
@@ -196,6 +199,20 @@ class IdeHaskellReplView
   setPrompt: (prompt) ->
     @promptDiv.innerText = prompt + '>'
 
+  getCompletions: (prefix) =>
+    return [] if prefix == '' or prefix == ' '
+    @ghci.sendCompletionRequest prefix
+    new Promise (resolve) =>
+      @completionHandler = @ghci.emitter.on 'complete', (lines) =>
```

Comment:
Also, you defined `onComplete`. Use it. As a rule of thumb, emitter-specific strings shouldn't leak outside the class which owns the emitter.